src/lib/Persistor.js

```diff
@@ -2,35 +2,50 @@
 
 var _ = require('underscore'),
     Q = require('q'),
-    Class = require('class.extend');
+    Class = require('class.extend'),
+    low = require('lowdb'),
+    FileSync = require('lowdb/adapters/FileSync'),
+    adapter, db;
 
 module.exports = Class.extend({
 
    init: function() {
-      // TODO: actually read/write to a file
-      this.inMemoryStorage = {};
+      adapter = new FileSync('./store/db.json');
+      db = low(adapter);
+      db.defaults({})
+         .write();
    },
 
    put: function(hash, range, value) {
-      if (!_.isObject(this.inMemoryStorage[hash])) {
-         this.inMemoryStorage[hash] = {};
+      var currentHash = db.get(hash).value(),
+          newObj = {};
+
+      if (!_.isObject(currentHash)) {
+         newObj = db.set(hash, {})
+            .write();
       }
 
-      this.inMemoryStorage[hash][range] = value;
+      newObj[range] = value;
+
+      db.get(hash)
+         .assign(newObj)
+         .write();
 
       return Q.when();
    },
 
    get: function(hash, range) {
-      if (_.isUndefined(range)) {
-         return Q.when(this.inMemoryStorage[hash]);
+      var currentHash = db.get(hash).value();
+
+      if (_.isUndefined(currentHash[range])) {
```

Comment:
I guess technically this is compatible  with the previous logic of `_.isUndefined(range)`. However, the result from the following will change with this implementation. Consider the output for this code:

```js
var Persistor = require('./lib/Persistor'),
    persistor = new Persistor();

persistor.put('games', 'first-game', { key: 'value' })
   .then(function() {
      return persistor.get('games', 'first-game').then(console.log);
   })
   .then(function() {
      return persistor.get('games').then(console.log);
   })
   .then(function() {
      return persistor.get('games', 'second-game').then(console.log);
   })
   .done();
```

For the current implementation, the following will be logged:

```
{ key: 'value' }
{ 'first-game': { key: 'value' } }
undefined
```

With this code, the output will be:

```
{ key: 'value' }
{ 'first-game': { key: 'value' } }
{ 'first-game': { key: 'value' } }
```

I think it makes more sense for `undefined` to be returned when there is no value found for the given `range`.

Reply:
Good point, I think I was confused as to the intent of this logic. In my latest update, I re-ordered the logic to check for the value from `range`, then look for the value returned by `currentHash`, otherwise return undefined. Does that seem right?